de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/toolbar/ToolbarViewPresenterImpl.java

```diff
@@ -394,4 +410,63 @@ public void onSuccess(String result) {
         });
     }
 
+    @Override
+    public void onAutomateHTPathlist() {
+        drFacade.getInfoTypes(new DataCallback<List<InfoType>>() {
+
+            @Override
+            public void onFailure(Integer statusCode, Throwable arg0) {
+                ErrorHandler.post(arg0);
+            }
+
+            @Override
+            public void onSuccess(List<InfoType> infoTypes) {
+                HTPathListAutomationDialog dialog =
+                        htPathListAutomationViewFactory.create(drSelectorFactory, infoTypes);
+                dialog.setSize(htAppearance.dialogWidth(), htAppearance.dialogHeight());
+                dialog.addOkButtonSelectHandler(event -> {
+                    if (dialog.isValid()) {
+                        HTPathListRequest request = dialog.getRequest();
+                        requestHTPathListCreation(dialog, request);
+                    } else {
+                        AlertMessageBox amb = new AlertMessageBox(htAppearance.heading(),
+                                                                  htAppearance.validationMessage());
+                        amb.show();
+                    }
+                });
+                dialog.addCancelButtonSelectHandler(event -> {
+                    dialog.hide();
+                });
+                dialog.show();
+            }
+        });
+
+    }
+
+    private void requestHTPathListCreation(HTPathListAutomationDialog dialog,
+                                           HTPathListRequest request) {
+        dialog.mask(htAppearance.processing());
+        drFacade.requestHTPathlistFile(request, new DataCallback<String>() {
+            @Override
+            public void onFailure(Integer statusCode, Throwable exception) {
+                ErrorHandler.post(htAppearance.requestFailed(), exception);
+            }
+
+            @Override
+            public void onSuccess(String result) {
+                dialog.hide();
+                IplantAnnouncer.getInstance()
```

Comment:
We should probably inject the `IplantAnnouncer` instead of calling `getInstance` here.